include/vigra/mathutil.hxx

```diff
@@ -164,6 +164,17 @@ VIGRA_DEFINE_MISSING_ABS(signed long long)
 using std::isinf;
 using std::isnan;
 
+#ifdef _LIBCPP_VERSION
+// On older versions of Mac, the system's libc++ does not provide isnan(double)
+// So we just provide it ourselves.
+template <typename T>
+typename std::enable_if<std::is_integral<T>::value, bool>::type
```

Comment:
>And if we need it (e.g. for reasons of genericity), why don't we just return true then?

I considered doing that, but I found [this page][1] which claims that it is theoretically possible for an architecture to specify integer Nan values (though x86 is not one of them).  Maybe it's not worth worrying about such bizarre cases, but using `x != x` should be more correct in such cases.  (The compiler will just optimize it away on most architectures.)

[1]: http://stackoverflow.com/questions/3949457/can-an-integer-be-nan-in-c